tools/stress/src/org/apache/cassandra/stress/Stress.java

```diff
@@ -196,4 +210,20 @@ public void run()
         }
     }
 
+    private static String threadDump(boolean lockedMonitors, boolean lockedSynchronizers) {
+        StringBuffer threadDump = new StringBuffer(System.lineSeparator());
+        ThreadMXBean threadMXBean = ManagementFactory.getThreadMXBean();
+        for(ThreadInfo threadInfo : threadMXBean.dumpAllThreads(lockedMonitors, lockedSynchronizers)) {
+            threadDump.append(threadInfo.toString());
+        }
+        return threadDump.toString();
+    }
+
+    private static void registerSignalHandler() {
+        SignalHandler handler = signal -> {
+            System.out.println("Caught signal: " + signal.getName());
+            //Thread dump will be handled by shutdown hook
+        };
+        Signal.handle(new Signal("ABRT"), handler);
```

Comment:
```suggestion
        Signal.handle(new Signal("ABRT"), handler);
        Signal.handle(new Signal("TERM"), handler);
        Signal.handle(new Signal("INT"), handler);
```

Reply:
I think there is no need to handle `SIGTERM` and `SIGINT`, since `Runtime.addShutdownHook` already handles them, either we do it through `Signal.handle` or use `Runtime.addShutdownHook`

Reply:
As suggested I removed shutdown hook and added those signals to the handler.